src/components/view/View.tsx

```diff
@@ -156,6 +156,63 @@ const getBoundaryCoordinatesForCircle = (
   };
 };
 
+const getBoundaryCoordinatesForRectangle = (
+  sourcePosition: Position,
+  sourceSize: Size,
+  destinationPosition: Position
+): Position => {
+  // If the two points are orthogonal, the boundary is half the width or height of the rectangle.
+  if (sourcePosition.x === destinationPosition.x) {
+    return {
+      x: sourcePosition.x,
+      y: sourceSize.height / 2
```

Comment:
This should be sourcePosition.y + sourceSize.height if destinationPosition.y is greater than sourcePosition.y and sourcePosition.y if destionationPosition.y is less than sourcePosition.y. I am able to reproduce in your repo this bug when I bring a node exactly perpendicular either horizontally or vertically.

Reply:
My bad. It should be sourcePosition.y + sourceSize.height/2 (if sourcePosition.y < destinationPosition.y) else sourcePosition.y - sourceSize.height/2 (if sourcePosition.y > destinationPosition.y)